catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/ResourceCacheInterface.java

```diff
@@ -39,8 +40,9 @@
      *
      * @param key
      * @return Resource obtained from cache
+     * @throws CacheException when no valid resource is found in the cache.
      */
-    Resource getValid(String key, Metacard latestMetacard);
+    Resource getValid(String key, Metacard latestMetacard) throws CacheException;
```

Comment:
Why do we throw an exception on a cache-miss?  This isn't an exceptional situation, it's the normal course of operation for a cache.